ecosystem/typescript/sdk/src/account/aptos_account.ts

```diff
@@ -126,6 +126,55 @@ export class AptosAccount {
     return HexString.fromUint8Array(hash.digest());
   }
 
+  /**
+   * Takes creator address and collection name and returns the collection object address.
+   * Collection object addresses are generated as sha256 hash of (creator address + collection_name)
+   *
+   * @param creator Creator address
+   * @param collectionName The collection name
+   * @returns The collection object address
+   */
+
+  static getCollectionObjectAddress(creator: MaybeHexString, collectionName: string): HexString {
+    const source = bcsToBytes(AccountAddress.fromHex(creator));
+    const seed = new TextEncoder().encode(collectionName);
+
+    const bytes = new Uint8Array([...source, ...seed, AuthenticationKey.DERIVE_OBJECT_FROM_SEED_SCHEME]);
+
+    const hash = sha3Hash.create();
+    hash.update(bytes);
+
+    return HexString.fromUint8Array(hash.digest());
+  }
+
+  /**
+   * Takes creator address, collection name and token name and returns the token object address.
+   * Token object addresses are generated as sha256 hash of (creator address + `collection's name + :: + token name`)
+   *
+   * @param creator Creator address
+   * @param collectionName The collection name
+   * @param tokenName The token name
+   * @returns The token object address
+   */
+
```

Comment:
I haven't seen the case where we leave a new line between doc block and code. Is this standard is TS/JS?

Reply:
haha not that I aware of - will fix